waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/DatabaseMappingUtils.java

```diff
@@ -0,0 +1,128 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.waggledance.mapping.service;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.apache.hadoop.hive.metastore.api.TableMeta;
+import org.slf4j.Logger;
+
+import com.hotels.bdp.waggledance.mapping.model.DatabaseMapping;
+import com.hotels.bdp.waggledance.util.Whitelist;
+
+public class DatabaseMappingUtils {
+
+  public static final String MANUAL_RESOLUTION_TYPE = "MANUAL";
+  public static final String PREFIXED_RESOLUTION_TYPE = "PREFIXED";
+  private static final String INTERRUPTED_MESSAGE = "Interrupted ";
+  private static final String SLOW_METASTORE_MESSAGE = "Metastore was slow to respond so results are omitted";
+  private static final String NULL_POINTER_MESSAGE = "Got nullpointer";
+
+  public static boolean isWhitelisted(String databasePrefix, String database, Map<String, Whitelist> mappedDbByPrefix) {
+    Whitelist whitelist = mappedDbByPrefix.get(databasePrefix);
+    if ((whitelist == null) || whitelist.isEmpty()) {
+      // Accept everything
+      return true;
+    }
+    return whitelist.contains(database);
+  }
+
+  public static List<String> getMappedWhitelistedDatabases(List<String> databases, DatabaseMapping mapping,
+                                                           Map<String, Whitelist> mappedDbByPrefix) {
+    List<String> mappedDatabases = new ArrayList<>();
+    for (String database : databases) {
+      if (DatabaseMappingUtils.isWhitelisted(mapping.getDatabasePrefix(), database, mappedDbByPrefix)) {
+        mappedDatabases.add(mapping.transformOutboundDatabaseName(database));
+      }
+    }
+    return mappedDatabases;
+  }
+
+  public static void shutdownExecutorService(ExecutorService executorService) {
+    executorService.shutdown();
+    try {
+      if (!executorService.awaitTermination(200, TimeUnit.MILLISECONDS)) {
+        executorService.shutdownNow();
+      }
+    } catch (InterruptedException e) {
+      executorService.shutdownNow();
+    }
+  }
+
+  private static List<?> getResultFromFuture(Iterator<DatabaseMapping> iterator, Future<List<?>> future)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    DatabaseMapping mapping = iterator.next();
+    long timeout = mapping.getTimeout();
+    return future.get(timeout, TimeUnit.MILLISECONDS);
+  }
+
+  public static void getDatabasesFromFuture(List<Future<List<?>>> futures, Iterator<DatabaseMapping> iterator,
```

Comment:
Is there a reason you pass `combined` ino this (and the other calls below) as opposed to returning a new List? (I suppose I'll find out as I read on ;) ) The reason I ask is that this is a side-effect of calling the method that isn't immediately obvious (and I think would be a functional programming no-no!) so if it's really necessary I'd suggest documenting it.

Reply:
Oh, you're right, this could just return a different list and then `combined` could get all of that input at the end!